### PR TITLE
Tests for mutableData permissions and forEach functions

### PR DESCRIPTION
**Pull request by bochaco**

### CODE REVIEW

File: src/native/_mutable.js [hunk 1]
```diff
@@ -31,6 +31,39 @@ function bufferLastEntry() {
         .concat([str, str.length]);
 }
 
+function callBackLastEntry() {
+  let fn = arguments[arguments.length - 1];
+  if (typeof fn !== 'function') throw Error('A function parameter _must be_ provided')
+
+  let cb = ffi.Callback("void", ['pointer', t.u8Pointer, t.usize],
```
Comment (gnunicorn) [inline src/native/_mutable.js]: you use this - among others -  for the for_each_value. But value has the callback signature `fn(*mut c_void, *const u8,  usize, u64),` - with the last one representing the version number. I think we need to wrap that again into the `{buf: Buffer, version: u64}` we've used other places.

File: src/native/_mutable.js [hunk 2]
```diff
@@ -194,13 +227,13 @@ module.exports = {
...
-    mdata_entries_for_each: Promisified(null, []),
+    mdata_entries_for_each: Promisified(callBackKeyValueLastEntry, []),
     mdata_entries_free: Promisified(null, []),
     mdata_keys_len: Promisified(null, t.usize),
-    mdata_keys_for_each: Promisified(null, []),
+    mdata_keys_for_each: Promisified(callBackLastEntry, []),
     mdata_keys_free: Promisified(null, []),
     mdata_values_len: Promisified(null, t.usize),
-    mdata_values_for_each: Promisified(null, []),
+    mdata_values_for_each: Promisified(callBackLastEntry, []),
```
Comment (gnunicorn) [inline src/native/_mutable.js]: we need to do something more with the callback to wrap here. see comment before.

File: test/mutuable_data.js [hunk 3]
```diff
@@ -187,9 +192,14 @@ describe('Mutable Data', () => {
         })
     );
 
-    it.skip('forEach on list of values', () => {
-      throw new Error('Test Not Implemented');
-    });
+    it('forEach on list of values', () => app.mutableData.newRandomPublic(TAG_TYPE)
+      .then((m) => m.quickSetup(TEST_ENTRIES).then(() => m.getValues()))
+      .then((values) => values.forEach((value) => {
```
Comment (gnunicorn) [inline test/mutuable_data.js]: this must expect value + version, too

File: test/mutuable_data.js [hunk 4]
```diff
@@ -340,47 +350,63 @@ describe('Mutable Data', () => {
...
-            )))
-    );
-
     it.skip('get permissions set', () => app.mutableData.newRandomPublic(TAG_TYPE)
         .then((m) => m.quickSetup(TEST_ENTRIES)
           .then(() => m.getPermissions()
             .then((perm) => app.auth.getPubSignKey()
-              .then((pk) => perm.getPermissionSet(pk.ref))
+              .then((pk) => perm.getPermissionSet(pk))
+              // the above command is failing with ERR_INVALID_SIGN_KEY_HANDLE
```
Comment (gnunicorn) [inline test/mutuable_data.js]: is this still failing?

File: src/native/_mutable.js (hunk 1 above)
Comment (bochaco) [inline src/native/_mutable.js]: Yes, you are right, I'll try to fix it today.

**Pull request merged**